config/generic-mks-sgenl-v2.cfg

```diff
@@ -0,0 +1,260 @@
+# This file contains common pin mappings for the MKS Sgen L V2
+# board. To use this config, the firmware should be compiled for the LPC1769.
+#klipper firmware configuration
+
+#   [*]enable extra low level configuration options
+#   micro-controller architecture (LPC176x (Smoothieboard))
+#   Processor model (lpc1769 (120 Mhz))
+#   []Target board uses Smoothieware bootloader
+
+[printer]
+kinematics: cartesian
+max_velocity: 500
+max_accel: 3000
+max_z_velocity: 25
+max_z_accel: 30
+
+
+[mcu]
+    serial:/dev/serial/by-id/usb-Klipper_lpc1769_0D80000EA29869AFD0E63F5EC72000F5-if00
+
+#########################################
+#  Stepper Settings
+#########################################
+[stepper_x]
+step_pin: P2.2
+dir_pin: P2.3
+enable_pin: !P2.1
+rotation_distance: 40
+microsteps: 16
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+endstop_pin: ^P1.29
+position_endstop: -10
+position_min: -10
+position_max: 263
+
+## Y Stepper
+[stepper_y]
+step_pin: P0.19
+dir_pin: P0.20
+enable_pin: !P2.8
+rotation_distance: 40
+microsteps: 16
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+endstop_pin: ^P1.27
+position_endstop: -20
+position_min: -20
+position_max: 220
+
+## Z0 Stepper
+[stepper_z]
+step_pin: P0.22
+dir_pin: P2.11
+enable_pin: !P0.21
+rotation_distance: 8
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+microsteps: 16
+endstop_pin: ^!P1.28
+position_endstop: 0
+position_min: -1.3
+position_max: 200
+
+## Z1 Stepper
+[stepper_z1]
+step_pin: P1.9
+dir_pin: P1.14
+enable_pin: !P0.10
+rotation_distance: 8
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+microsteps: 16
+
+########################################
+# TMC2209 configuration
+########################################
+[tmc2209 stepper_x]
+uart_pin: P1.1
+run_current: 0.7
+sense_resistor: 0.110
+
+[tmc2209 stepper_y]
+uart_pin: P1.8
+run_current: 0.7
+sense_resistor: 0.110
+
+[tmc2209 stepper_z]
+uart_pin: P1.10
+run_current: 0.7
+sense_resistor: 0.110
+
+[tmc2209 stepper_z1]
+uart_pin: P1.17
+run_current: 0.7
+sense_resistor: 0.110
+
+[tmc2209 extruder]
+uart_pin: P1.15
+run_current: 0.3
+sense_resistor: 0.110
+
+#[tmc2209 extruder1]
+#uart_pin: P1.17
+#run_current: 0.3
+#sense_resistor: 0.110
+
+########################################
+# TMC2208 configuration
+########################################
+#[tmc2208 stepper_x]
+#uart_pin: P1.1
+#run_current: 0.7
+#
+#[tmc2208 stepper_y]
+#uart_pin: P1.8
+#run_current: 0.7
+#
+#[tmc2208 stepper_z]
+#uart_pin: P1.10
+#run_current: 0.65
+#
+#[tmc2208 extruder]
+#uart_pin: P1.15
+#run_current: 0.7
+#
+#[tmc2208 extruder1]
+#uart_pin: P1.17
+#run_current: 0.7
```

Comment:
Is there a special reason to duplicate the TMC UART conf? I'd say an example showing the UART conf and one showing the SPI conf is perfect.
Goal is to provide new users the needed pin information as not everybody can identify this information from the schematics or pin-out.

Reply:
i think i'm following you now.

would something like this suffice ?
```
#[tmc2209 stepper_x]
#[tmc2130 stepper_x]
#[tmc2208 stepper_x]
#uart_pin: P1.1
#run_current: 0.7
#sense_resistor: 0.110
### enable below if using stepper in SPI mode 
#cs_pin: P1.1
#spi_software_miso_pin: P0.5
#spi_software_mosi_pin: P1.16
#spi_software_sclk_pin: P0.4
#run_current: 0.7
#diag1_pin: P1.29
```

Reply:
To keep in line with the existing cfg, I would do something like:

```
########################################
# TMC UART configuration, e.g. tmc2208 
########################################

#[tmc2208 stepper_x]
#uart_pin: XX
#run_current: 0.8

... # for all potential axes

########################################
# TMC SPI configuration, e.g. tmc2130 
########################################

#[tmc2130 stepper_x]
#cs_pin: xx
#spi_software_miso_pin: xx
#spi_software_mosi_pin: xx
#spi_software_sclk_pin: xx
#diag1_pin: xx
#run_current: 0.8

... # for all potential axes
```

The most critical thing for new users is identifying the respective pins and the rest can easily identified via the Klipper documentation.

Reply:
since the 2208 drivers and 2009 drivers are basically the same, wouldn't it be better to have it like this to save a few lines ?
but then it would be better to keep it all the same as other configs.
```
########################################
# TMC2209 or TMC2208 configuration
########################################
[tmc2209 stepper_x]
#[tmc2208 stepper_x]
uart_pin: P1.1
run_current: 0.7
sense_resistor: 0.110

```

Reply:
Yes, basically I like this approach but it has certain drawbacks:
* It works for the UART types 2208 / 2209 / 2224 --> No need specify `sense_resistor` as it has a matching default value
* The SPI types 2660 / 5160 are more difficult --> 5160 has a default `sense_resistor` whereas 2660 has not. Here we would have to add a note to set the correct `sense_resistor` value

Reply:
> Yes, basically I like this approach but it has certain drawbacks:
> 
>     * It works for the UART types 2208 / 2209 / 2224 --> No need specify `sense_resistor` as it has a matching default value
> 
>     * The SPI types 2660 / 5160 are more difficult --> 5160 has a default `sense_resistor` whereas 2660 has not. Here we would have to add a note to set the correct `sense_resistor` value

since the 2208/2209 and basically the same, i can format it like that.
if a user has the 2660/5160 drivers can we assume they will know how to edit the config themselves ?

Reply:
>  a user has the 2660/5160 drivers can we assume they will know how to edit the config themselves ?

Generally, we should deliver both examples (UART & SPI), as they document the needed board pin, which is biggest need in front of the users. My personal opinion would be to do it comprehensively, e.g. by adding a comment to `sense_resistor` for the SPI variants.

Apart from this, my view is that the users buy what sounds cool on Ali. I have seen people driving a 300 mA stepper with a 5160 🤷 

To resolve the build failure, please try changing
```
[mcu]
    serial:/dev/serial/by-id/usb-Klipper_lpc1769_0D80000EA29869AFD0E63F5EC72000F5-if00
```
to
```
[mcu]
    serial:/dev/serial/by-id/usb-Klipper_Klipper_firmware_12345-if00
```

Reply:
Damn, sorry my fault: White space missing

Reply:
great thanks for the help!!

Reply:
> Damn, sorry my fault: White space missing

would it be worth while changing the klipper tests to see if there is no "indentation" in from of "serial:" ?